src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -107,22 +105,68 @@ public void RecordParsing01()
                     }
                     N(SyntaxKind.SemicolonToken);
                 }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+
+            // In langversion 8, this is a method
+            UsingTree(text, options: TestOptions.Regular8,
```

Comment:
Are there any tests that show the parsing "variants" inside namespaces and/or types? It would be nice if we also get a FeatureInPreview diagnostic there. I'm guessing parsing will just let it through and binding will fail to find a type named 'record'.

I feel like if it's not too difficult to achieve we should try to make sure the workflow of "install VS preview, pop open a C# file in my existing project, and just start using the preview feature" will work. Even something that just notices when a method that returns an ErrorType with the identifier 'record' and adds an additional FeatureInPreview diagnostic seems adequate.

Also, is there any chance that syntax nodes will be reused when changing the LangVersion in an open project? I'm guessing no because today you have to re-parse to get the right diagnostics, but figured it was worth checking.

Reply:
It would certainly be useful to get the "FeatureInPreview" diagnostic but I'm going to say that's out-of-scope for this PR.

> Also, is there any chance that syntax nodes will be reused when changing the LangVersion in an open project?

Nope, changing the parse options always requires reparsing every syntax tree from scratch.